definitions/features/foreman_proxy.rb

```diff
@@ -34,16 +34,17 @@ def find_http_error_msg(array_output, curl_http_status)
         next unless str.include?('HTTP')
         http_line = str
       end
+      return http_line if http_line.empty?
       http_line.split(curl_http_status.to_s).last.strip
     end
   end
 
   def run_dhcp_curl
     curl_resp = execute(dhcp_curl_cmd)
-    array_output = curl_resp.split(/\r\n/)
+    array_output = curl_resp.split(/\r?\n/)
     result_array = array_output.last.split(/\n/)
     curl_http_status = result_array.delete_at(result_array.length - 1).strip.to_i
-    curl_http_resp = parse_json(result_array.join(''))
+    curl_http_resp = parse_json(array_output[0])
```

Comment:
Just in case, shouoln't we also check that `array_output` is not empty?

Reply:
@iNecas, yes we can.
But array_output contains result and http status code. Result can be empty but array_output will always have http status code.
For empty result + status code 200, we are raising failure.

Reply:
ok